### PR TITLE
Functional Tests : BO - Left Menu

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : BO - Left Menu
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Related PRs       | N/A
| How to test?      | Run command `HEADLESS=false TEST_PATH="functional/BO/15*/01*" URL_FO="https://domain.tld/" npm run test:specific`
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/pages/BO/BObasePage.js [hunk 1]
```diff
@@ -278,18 +383,112 @@ class BOBasePage extends CommonPage {
...
+    return (await page.$$(`${linkSelector}.link-active`)).length > 0;
+  }
+
+  /**
+   * Return is the navbar is visible
+   * @param page {Page} Browser tab
+   * @return {Promise<boolean>}
+   */
+  async isNavbarVisible(page) {
+    return this.elementVisible(page, '.nav-bar:not(.mobile-nav)', 1000);
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/BObasePage.js]: Please create a new variable for the selector

File: tests/UI/pages/BO/BObasePage.js [hunk 2]
```diff
@@ -278,18 +383,112 @@ class BOBasePage extends CommonPage {
...
+  }
+
+  /**
+   * Return is the navbar is visible
+   * @param page {Page} Browser tab
+   * @return {Promise<boolean>}
+   */
+  async isMobileMenuVisible(page) {
+    return this.elementVisible(page, '.js-mobile-menu', 1000);
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/BObasePage.js]: Same here

File: tests/UI/pages/BO/BObasePage.js [hunk 3]
```diff
@@ -278,18 +383,112 @@ class BOBasePage extends CommonPage {
...
+    }
+  }
+
+  /**
+   * Returns if the sidebar is collapsed
+   * @param page {Page} Browser tab
+   * @return {Promise<boolean>}
+   */
+  async isSidebarCollapsed(page) {
+    return this.elementVisible(page, 'body.page-sidebar-closed', 1000);
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/BObasePage.js]: same here

File: tests/UI/pages/BO/BObasePage.js [hunk 4]
```diff
@@ -278,18 +383,112 @@ class BOBasePage extends CommonPage {
...
+   * @return {Promise<boolean>}
+   */
+  async setSidebarCollapsed(page, isCollapsed) {
+    const isCurrentCollapsed = this.isSidebarCollapsed(page);
+    if (isCurrentCollapsed !== isCollapsed) {
+      await Promise.all([
+        page.click(this.menuCollapse),
+        this.waitForVisibleSelector(
+          page,
+          isCollapsed ? 'body.page-sidebar-closed' : 'body:not(.page-sidebar-closed)',
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/BObasePage.js]: same here

File: tests/UI/pages/commonPage.js [hunk 5]
```diff
@@ -478,6 +478,21 @@ class CommonPage {
...
+   * @param mobileSize {boolean} Define if the viewport is for mobile or not
+   * @returns {Promise<void>}
+   */
+  async resize(page, mobileSize) {
+    if (mobileSize) {
+      await page.setViewportSize({width: 600, height: 600});
+    } else {
+      await page.setViewportSize({width: global.BROWSER.width, height: global.BROWSER.height});
+    }
+    await this.waitForSelector(page, '.js-mobile-menu', mobileSize ? 'visible' : 'hidden');
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/commonPage.js]: same here

**Review by nesrineabdmouleh — COMMENTED** [review]
Good job @Progi1984  :muscle: 
Just a small feedback

File: tests/UI/pages/BO/BObasePage.js (hunk 1 above)
Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js]: Fixed

File: tests/UI/pages/BO/BObasePage.js (hunk 2 above)
Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js]: Fixed

File: tests/UI/pages/BO/BObasePage.js (hunk 3 above)
Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js]: Fixed

File: tests/UI/pages/BO/BObasePage.js (hunk 4 above)
Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js]: Fixed

File: tests/UI/pages/commonPage.js (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/commonPage.js]: Fixed

**Comment by nesrineabdmouleh** [comment]
@Progi1984 Thank you for your fixes :)
All is good for me but you have a conflicting file

**Comment by Progi1984** [comment]
@nesrineabdmouleh It's fixed

**Review by nesrineabdmouleh — DISMISSED** [review]

File: tests/UI/pages/BO/BObasePage.js [hunk 6]
```diff
@@ -703,6 +707,21 @@ class BOBasePage extends CommonPage {
...
     await page.waitForNavigation('networkidle');
   }
+
+  /**
+   * Resize the page to defined viewport
+   * @param page {Page} Browser tab
+   * @param mobileSize {boolean} Define if the viewport is for mobile or not
+   * @returns {Promise<void>}
+   */
+  async resize(page, mobileSize) {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/BObasePage.js:722]: Resize should be a `commonPage` function that we call on BO and FO
```js 
if (mobileSize) {
  await page.setViewportSize({width: 600, height: 600});
} else {
  await page.setViewportSize({width: global.BROWSER.width, height: global.BROWSER.height});
}

Comment (boubkerbribri) [reply tests/UI/pages/BO/BObasePage.js:722]: Than define resize on`BoBasePage` maybe

Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js:722]: @boubkerbribri No, because we wait for a selector only available on backoffice

Comment (Progi1984) [reply tests/UI/pages/BO/BObasePage.js:722]: @boubkerbribri & @nesrineabdmouleh Fixed.

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Pull request merged**